### ISSUE TITLE
Allow aria-label to be set on the canvas

### ISSUE THREAD

**Issue by rsids** [openlayers/openlayers#12485]
To give visually impaired users context for the canvas element, it should be possible to set an `aria-label` on the canvas created by OpenLayers.
By setting the `aria-label` we can comply to [WCAG 1.1.1: Non-text-content](https://www.w3.org/WAI/WCAG21/Understanding/non-text-content.html).
The easiest way to achieve this is by simply hardcoding the aria-label to `aria-label="Canvas element marked as an image"`,
but it would be better to be able to set it via the options in the constructor of the map:
```javascript
const map = new Map({
    ariaLabel: "An interactive map of the world",
    ...
})
```
This way the user has control over what the label should read.

**Comment by tschaub** [comment]
Is there a drawback to setting the label on the target element used by the map?

```html
<div id=“map” aria-label=“Interactive map”></div>
```

```js
new Map({target: 'map'});
```

**Comment by Malvoz** [comment]
This relates to https://github.com/openlayers/openlayers/issues/12314 (which does not use `aria-label`, such that `aria-label` is available for developers to uniquely describe each map, if they wish).

> Is there a drawback to setting the label on the target element used by the map?

I would say there's a drawback to setting it on the `canvas` element. Setting it on the target element instead gives context to all children of the element, i.e. it gives context to other elements such as the controls' container.

**Comment by krechj** [comment]
We could use an aria-label on the zoom-in and zoom-out buttons. So visually impaired people can understand them. This would help us to comply to _WCAG SuccessCriterium 2.4.6. Headings and Labels_

As rsids said: a hardcoded aria-label would go a long way, but the ability to set it in the configurations would be better (preferably we need to set the labels in Dutch).

**Comment by nikolas** [comment]
This issue is still valid, and shouldn't be marked as stale.

**Comment by ahocevar** [comment]
@nikolas It would help if you could add a pull request that shows what exactly you have in mind.

**Comment by tschaub** [comment]
> @nikolas It would help if you could add a pull request that shows what exactly you have in mind.

Agreed.  And if not a pull request, please provide example code that you would like to see working.
